File: citasService/src/auth0backend.js
import axios from 'axios'

const getRole = async (req) => {
    const accessToken = req.headers.authorization.split(' ')[1];
    if (!accessToken) return 'Unauthorized';
    try {
        const user = await axios.get(`https://${process.env.DOMAIN}/userinfo`, {
            headers: {
                'Authorization': 'Bearer ' + accessToken
            }
        });
        return user['data']['roles/'][0];
    } catch (error) {
        console.log(error);
        return 'Unauthorized';
    }
}

export default getRole;

File: usersService/users/auth0backend.py
import requests
from django.conf import settings

def getRole(request):
    access_token = request.headers['Authorization']
    access_token = access_token.split(" ")[1]
    try:
        user = requests.get(settings.AUTH_PATH, headers={"Accept":"*/*", "Authorization": "Bearer " + access_token})
        user = user.json()
    except Exception as e:
        print(e)
        return "Unauthorized"
    
    return user['roles/'][0]